js/exporter/jspdf/export.js

```diff
@@ -0,0 +1,181 @@
+                        const cellStyle = styles[dataProvider.getStyleId(rowIndex, cellIndex)];
+                        const gridCell = cellData.cellSourceData;
+                        const pdfCell = {
+                            content: cellData.value,
+                            styles: {}
+                        };
+
+                        this.assignCellStyle(pdfCell, gridCell, columns[cellIndex], cellStyle);
+
+                        rowType = rowType || gridCell.rowType;
+                        if(rowType === 'header') {
+                            if(pdfCell.content !== '') { row.push(pdfCell); }
+                        } else {
+                            row.push(pdfCell);
+                        }
+                    }
+
+                    this.addRow(autoTableOptions, rowType, row);
+                }
+
+                jsPDFDocument.autoTable(autoTableOptions);
+
+                resolve(autoTableOptions);
+            }).always(() => {
+
+            });
+        });
+    },
+
+    addRow: function(autoTableOptions, rowType, row) {
+        if(rowType === 'header') { autoTableOptions.head.push(row); }
+        if(rowType === 'data') { autoTableOptions.body.push(row); }
+    },
+
+    assignCellStyle: function(pdfCell, gridCell, column, cellStyle) {
+        if(gridCell.rowType === 'header') {
+            // eslint-disable-next-line spellcheck/spell-checker
+            if(column.alignment) { pdfCell.styles.halign = column.alignment; }
+        } else {
+            if(cellStyle) {
+                // eslint-disable-next-line spellcheck/spell-checker
+                if(cellStyle.alignment) { pdfCell.styles.halign = cellStyle.alignment; }
+                if(cellStyle.bold) { pdfCell.styles.fontStyle = 'bold'; }
+                if(cellStyle.wrapText) { pdfCell.styles.cellWidth = 'wrap'; }
+                if(cellStyle.dataType === 'date') { pdfCell.content = dateLocalization.format(new Date(pdfCell.content), cellStyle.format); }
+            }
+        }
+    },
+
+    setTableWidth: function(autoTableOptions, width) {
+        if(width) { autoTableOptions.tableWidth = this.convertUnitToPoints(width, 'px'); }
+    },
+
+    setColumnWidths: function(autoTableOptions, columns, keepColumnWidths) {
+        const columnStyles = autoTableOptions.columnStyles;
+
+        for(let i = 0; i < columns.length; i++) {
+            columnStyles[i] = columnStyles[i] || {};
+
+            const columnWidth = columns[i].gridColumn.width;
+            if(keepColumnWidths && (typeof columnWidth === 'number') && isFinite(columnWidth)) {
+                columnStyles[i].cellWidth = this.convertUnitToPoints(columnWidth, 'px');
+            } else {
+                columnStyles[i].cellWidth = 'auto';
+            }
+        }
+    },
+
+    getRelativeProportionsToPt: function(unit) {
+        // Unit table from https://github.com/MrRio/jsPDF/blob/ddbfc0f0250ca908f8061a72fa057116b7613e78/jspdf.js#L791
```

Comment:
Do not copy paste outer code!
Make some changes to avoid warnings at least from automatic code comparers.
Our code should not look like "simply copied".